origami_vs:
  
  goal_pose_x: 450
  goal_pose_y: 300

  control_rate: 30

  lambda: 2

  init_max_length_hi: 0
  init_max_length_lo: 180
  init_min_length_hi: 0
  init_min_length_lo: 70

  trans_ee_marker_y: -25
  trans_ee_marker_z: 25
  trans_base_marker_y: 0
  trans_base_marker_z: -54

  origami_module_d: 40

  ratio_mm_pixel: 0.4

origami_adaptive_vs:
  goal_pose: [250, 300]
  lambda: 2
  control_rate: 15
  window_size: 200
  gamma: 1.0e-7
  epsilon: 1.0e-5
  update_threshold: 2
  init_jacobian: [.1,-.3,.7,-.11]    # initialize non-zero combined Jacobian
  saturation: 25

origami_skeleton_vs:
  # VS PARAMS
  # lambda: [0.45, 0.25, 0.45, 0.45, 0.45]
  # lambda: [1.25, 1.25, 2.0, 2.0, 2.5, 2.5, 2.5, 2.5]
  lambda: [1.75, 1.75, 0.5, 1.75, 1.75, 0.5, 2.5, 2.5, 0.5]
  # lambda: [4.0, 2.5, 5.0, 2.5, 5.0, 3.5, 5.0, 3.5]
  # lambda: [0.1, 0.1]
  # lambda: [0.75, 0.75, 1.0, 1.0]
  control_rate: 10
  
  # CURVE FIT PARAMS
  curve_type: clothoid

  # feature_type: points
  feature_type: kappa
  # feature_type: curve_parameters
  no_of_points: 3       # Use this if using kappa as feature type
  no_of_features: 9        
  # no_of_actuators: 4  # when using 2 modules
  no_of_actuators: 5    # when using finger with 2 modules
  no_of_modules: 2
  num_of_segments: 1        # Probably not needed
  degree: 2                 # Probably not needed
  finger: Yes
  # curve_type: spline
  # no_of_actuators: 4
  # no_of_modules: 2
  # num_of_segments: 2
  # degree: 2
  # no_of_features: 6

  # ESTIMATION & ADAPTIVE UPDATE PARAMS
  gamma: 1.0e-7
  window_size: 15
  update_threshold: 10
  saturation: 10
  epsilon: 1.0e-5

########### 9 FEATURES #################
  init_jacobian: [0.1,-0.3,0.7,-0.11,
                  0.13,0.19,-0.17, 0.23,
                  0.29,0.37,-0.31,0.41,
                  0.17,0.11,-0.41, 0.1,
                  0.3,0.17,-0.23, 0.29,
                  0.47,0.31,-0.37, 0.7,
                  0.37, 0.39, 0.11, -0.43,
                  -0.19, 0.41, 0.53, 0.17,
                  0.11, -0.23, 0.29, 0.51]

########### 10 FEATURES #################
  # init_jacobian: [0.1,-0.3,0.7,-0.11,
  #                 0.13,0.19,-0.17, 0.23,
  #                 0.29,0.37,-0.31,0.41,
  #                 0.17,0.11,-0.41, 0.1,
  #                 0.3,0.17,-0.23, 0.29,
  #                 0.47,0.31,-0.37, 0.7,
  #                 0.37, 0.39, 0.11, -0.43,
  #                 -0.19, 0.41, 0.53, 0.17,
  #                 0.11, -0.23, 0.29, 0.51,
  #                 0.53, 0.29, 0.71, 0.67]

########### 8 FEATURES ##################
  # init_jacobian: [0.1,-0.3,0.7,-0.11,
  #                 0.13,0.19,-0.17, 0.23,
  #                 0.29,0.37,-0.31,0.41,
  #                 0.17,0.11,-0.41, 0.1,
  #                 0.3,0.17,-0.23, 0.29,
  #                 0.47,0.31,-0.37, 0.7,
  #                 0.37, 0.39, 0.11, -0.43,
  #                 -0.19, 0.41, 0.53, 0.17]

############ 5 FEATURES ##################
  # init_jacobian: [0.1,-0.3,0.7,-0.11,
  #                 0.13,0.19,-0.17, 0.23,
  #                 0.29,0.37,-0.31,0.41,
  #                 0.17,0.11,-0.41, 0.1,
  #                 0.3,0.17,-0.23, 0.29]

############ 4 FEATURES ##################
  # init_jacobian: [0.1,-0.3,0.7,-0.11,
  #                 0.13,0.19,-0.17, 0.23,
  #                 0.29,0.37,-0.31,0.41,
  #                 0.17,0.11,-0.41, 0.1]

############ 3 FEATURES ###################
  # init_jacobian: [0.1,-0.3,0.7,-0.11, 0.13, 0.19,-0.17, 0.23, 0.29, 0.37, -0.31, 0.41]    # initialize non-zero combined Jacobian


origami_sim:
# Robot config
  no_of_features: 3
  no_of_actuators: 3

# Control parameters
  control_rate: 50
  lam: 30
  feedforward: Yes

# Target settings
  target_feature: [250, 60, 0]
  stationary_target: No
  radius: 30
  slope: 200
  target_speed: 0.05
  target_range: 6.28

origami_sim_clothoid_vs:
  no_of_features: 3
  feature_type: clothoid